packages/itwin/tree-widget/src/components/trees/imodel-content-tree/IModelContentTreeDefinition.ts

```diff
@@ -369,46 +362,26 @@ export class IModelContentTreeDefinition implements HierarchyDefinition {
     return defs;
   }
 
-  private async createCategoryChildrenQuery({
-    parentNodeInstanceIds: categoryIds,
-    parentNode,
-    instanceFilter,
-    viewType,
-  }: DefineInstanceNodeChildHierarchyLevelProps & { viewType: "2d" | "3d" }): Promise<HierarchyLevelDefinition> {
+  private async createCategoryChildrenQuery(props: DefineInstanceNodeChildHierarchyLevelProps & { viewType: "2d" | "3d" }): Promise<HierarchyLevelDefinition> {
+    const { parentNodeInstanceIds: categoryIds, parentNode, instanceFilter, viewType } = props;
+    const modelIds = parseIdsSelectorResult(parentNode.extendedData?.modelIds);
+
     // We only want to handle a category added as a child of `GeometricModel2d` or `GeometricModel3d`.
-    if (!parentNode.extendedData?.isCategory) {
-      return [];
-    }
-    const { elementClass, modelClass } = getClassNameByViewType(viewType);
-    const modelIds: Id64String[] =
-      parentNode.extendedData && parentNode.extendedData.hasOwnProperty("modelIds") && Array.isArray(parentNode.extendedData.modelIds)
-        ? parentNode.extendedData.modelIds.reduce(
-            (arr, ids: Id64String | Id64String[]) => [...arr, ...(Array.isArray(ids) ? ids : [ids])],
-            new Array<Id64String>(),
-          )
-        : [];
     if (modelIds.length === 0) {
-      throw new Error(`Invalid category node "${parentNode.label}" - missing model information.`);
+      return this.createElementChildrenQuery(props);
     }
```

Comment:
please add a comment explaining this branch

Reply:
Improved the comment above